formula/src/test/java/com/instacart/formula/subjects/StartStopFormula.kt

```diff
@@ -40,13 +42,16 @@ class StartStopFormula(runtime: TestableRuntime) : Formula<Unit, State, Output>
             },
             output = Output(
                 state = state.count,
-                startListening = context.onEvent {
-                    transition(state.copy(listenForEvents = true))
-                },
-                stopListening = context.onEvent {
-                    transition(state.copy(listenForEvents = false))
-                }
+                // We need to specify keys since `UpdateListenFlag` type is used two times.
+                startListening = context.onEvent("start", UpdateListenFlag(listen = true)),
+                stopListening = context.onEvent("stop", UpdateListenFlag(listen = false)),
             )
         )
     }
+
+    private class UpdateListenFlag(val listen: Boolean): Transition<State, Unit> {
```

Comment:
Example of using separate type for `Transition`

Reply:
Would be interesting to add transition specific unit tests.

Reply:
Sorry, not sure I understand this fully. Could you explain a bit more?

Reply:
I meant you could create unit tests specifically for transitions instead of for the whole formula if you create classes that encapsulate transitions.

Reply:
Exactly, though from my experience, I would still strongly suggest testing the whole formula as those tests tend to be more robust.